src/uct/ib/dc/dc_mlx5.c

```diff
@@ -900,6 +904,8 @@ uct_dc_mlx5_iface_dcis_create(uct_dc_mlx5_iface_t *iface,
     iface->tx.bb_max = dci->txwq.bb_max;
     uct_dc_mlx5_destroy_dci(iface, dci);
 
+    ucs_array_length(&iface->tx.dcis) = 0;
```

Comment:
why needed?

Reply:
because dci 0 was destroyed and that place is effectively empty